code/API_definitions/qos-booking.yaml

```
@@ -0,0 +1,1306 @@
+tags:
+  - name: QoS Booking
+    description: Manage the booking of QoS
+
+paths:
+  /device-qos:
+    post:
+      tags:
+        - QoS Booking
+      summary: Sets a new booking of QoS for a device
+      description: |
+        Triggers a new booking to assign certain QoS Profile to certain device.
+
+        - If the booking is completed synchronously, the response will be 201 with `status` = `SCHEDULED`.
+        - If the booking request is accepted but not yet completed, the response will be 201 with `status` = `REQUESTED`.
+        - If the operator determines synchronously that the booking request cannot be fulfilled, the response will be 201 with `status` = `UNAVAILABLE`.
+
+        - If the request includes  the `sink` and `sinkCredential` properties, the client will receive a `status-changed` event with the outcome of the process. The event will be sent also for synchronous operations.
+
+        **NOTES:**
+        - When the booking status becomes `UNAVAILABLE`, the QoS Booking resource is not immediately released, but will get deleted automatically, at earliest 360 seconds after.
+
+        This behavior allows clients which are not receiving notification events but are polling, to get the booking status information.
+        - The access token may be either 2-legged or 3-legged.
+          - If a 3-legged access token which is associated with a device is used, it is recommended NOT to include the `device` parameter in the request (see "Handling of device information" within the API description for details).
+          - If a 2-legged access token is used, the device parameter must be provided and identify a device.
+
+      operationId: createBooking
+      parameters:
+        - $ref: "#/components/parameters/x-correlator"
+      requestBody:
+        description: Parameters to create a new booking
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CreateBooking"
+        required: true
+      callbacks:
+        notifications:
+          "{$request.body#/sink}":
+            post:
+              summary: Booking notifications callback
+              description: |
+                Important: this endpoint is to be implemented by the API consumer.
+                The QoS Booking server will call this endpoint whenever any QoS Booking change related event occurs.
+                Currently only `Booking_STATUS_CHANGED` event is defined.
+              operationId: postBookingNotification
+              parameters:
+                - $ref: "#/components/parameters/x-correlator"
+              requestBody:
+                required: true
+                content:
+                  application/cloudevents+json:
+                    schema:
+                      $ref: "#/components/schemas/CloudEvent"
+                    examples:
+                      Booking_STATUS_CHANGED_EXAMPLE:
```

```suggestion
                      BOOKING_STATUS_CHANGED_EXAMPLE:
```

To be applied in all spec: Booking_ -> BOOKING_